deltacat/compute/compactor_v2/deletes/utils.py

```diff
@@ -0,0 +1,157 @@
+            assert len(delete_columns) > 0, "At least 1 delete column is required"
+            # delete columns should exist in underlying table
+            delete_dataset = params.deltacat_storage.download_delta(
+                delete_delta,
+                file_reader_kwargs_provider=params.read_kwargs_provider,
+                columns=delete_columns,
+                storage_type=StorageType.LOCAL,
+                max_parallelism=1,
+                **params.deltacat_storage_kwargs,
+            )
+            consecutive_delete_tables.extend(delete_dataset)
+        delete_table: pa.Table = pa.concat_tables(consecutive_delete_tables)
+        delete_file_envelope: DeleteFileEnvelope = DeleteFileEnvelope.of(
+            start_stream_position,
+            delta_type=DeltaType.DELETE,
+            table=delete_table,
+            delete_columns=delete_columns,
+        )
+        delete_file_envelopes.append(delete_file_envelope)
+    return delete_file_envelopes
+
+
+def prepare_deletes(
+    params: CompactPartitionParams,
+    input_deltas: List[Delta],
+    *args,
+    **kwargs,
+) -> PrepareDeleteResult:
+    """
+    Prepares delete operations for a compaction process.
+    This function processes all the annotated deltas and consolidates consecutive DELETE deltas.
+    It creates a range dictionary of these consolidated delete operations of the earliest stream position to the Ray obj references to the delete table
+    Additionally, non-DELETE deltas are accumulated in a separate list.
+
+    Args:
+        params (CompactPartitionParams): Parameters for the compaction process.
+        uniform_deltas (List[Delta]): A list of Delta objects representing
+            delete operations.
+
+    Returns:
+        Tuple[List[Delta], IntegerRangeDict]:
+            - A list of Annotated Deltas excluding all non-delete operations.
+            - A dictionary (IntegerRangeDict) containing consolidated delete operations, where the keys
+            are the earliest stream positions of the consolidated delete operations, and the values
+            are Ray object references to PyArrow Tables representing the consolidated delete tables.
+            If there are no delete operations, this dictionary will be empty
+
+    Raises:
+        AssertionError: If a delete operation does not have the required properties defined.
+    """
+    if not input_deltas:
+        return PrepareDeleteResult(input_deltas, [])
+    assert all(
+        input_deltas[i].stream_position <= input_deltas[i + 1].stream_position
+        for i in range(len(input_deltas) - 1)
+    ), "Uniform deltas must be in non-decreasing order by stream position"
+    start_stream_spos_to_delete_delta_sequence: Dict[
+        int, List[Delta]
+    ] = _aggregate_delete_deltas(input_deltas)
+    delete_file_envelopes: List[DeleteFileEnvelope] = _get_delete_file_envelopes(
+        params, start_stream_spos_to_delete_delta_sequence
+    )
```

Comment:
As a safety measure, assert the length of `start_stream_spos_to_delete_delta_sequence` and `delete_file_envelopes` to be equal.